test/Microsoft.ServiceHub.Framework.Tests/ServerFactoryTests.cs

```diff
@@ -13,6 +15,69 @@ public ServerFactoryTests(ITestOutputHelper logger)
 	{
 	}
 
+	[Fact]
+	public async Task TestAsyncFactoryCanConnect()
+	{
+		TaskCompletionSource<Stream> serverStreamSource = new();
+		Stream? clientStream = null;
+		IIpcServer server = ServerFactory.Create(
+			stream =>
+			{
+				serverStreamSource.TrySetResult(stream);
+				return Task.CompletedTask;
+			},
+			new ServerFactory.ServerOptions
+			{
+				TraceSource = this.CreateTestTraceSource(nameof(this.TestAsyncFactoryCanConnect)),
+			});
+		try
+		{
+			clientStream = await ServerFactory.ConnectAsync(server.Name, default, this.TimeoutToken);
+			await serverStreamSource.Task.WithCancellation(this.TimeoutToken);
+		}
+		finally
+		{
+			await server.DisposeAsync();
+		}
+
+		// Now verify that the pipe still works, since we disposed the server.
+		Task writeTask = clientStream.WriteAsync(new byte[] { 1, 2, 3 }, 0, 3, this.TimeoutToken);
+		byte[] buffer = new byte[3];
+		Stream serverStream = await serverStreamSource.Task.WithCancellation(this.TimeoutToken);
+		Task<int> bytesReadTask = serverStream.ReadAsync(buffer, 0, 3, this.TimeoutToken);
+		await writeTask.WithCancellation(this.TimeoutToken);
+		Assert.NotEqual(0, await bytesReadTask.WithCancellation(this.TimeoutToken));
+	}
+
+	[Fact]
+	public async Task TestAsyncFactoryNoSpin()
+	{
+		// Stopwatch to measure wall-clock time (total elapsed time)
+		var stopwatch = new Stopwatch();
+		stopwatch.Start();
+
+		// Get the current process to measure CPU usage
+		var process = Process.GetCurrentProcess();
+		TimeSpan initialCpuTime = process.TotalProcessorTime;
+
+		// Try to connect to non-existent pipe, cancel after some time
+		try
+		{
+			var cts = new CancellationTokenSource(5 * 1000); // Delay for 5 seconds
+			await ServerFactory.ConnectAsync("NonExistentPipe", default, cts.Token);
+		}
+		catch (TaskCanceledException)
+		{
+			// Ignore exception
+		}
```

Comment:
The way you have this written, if no exception is thrown at all, the test may still pass.
If you expect an exception, use syntax that fails when it doesn't occur.

```suggestion
		var cts = new CancellationTokenSource(5 * 1000); // Delay for 5 seconds
		await Assert.ThrowsAsync<TaskCanceledException>(() => ServerFactory.ConnectAsync("NonExistentPipe", default, cts.Token));
```

Reply:
Good point. I only want an exception to be thrown if it's not a `TaskCancelledException`. I expect the exception to be thrown, so I'll add a check to verify it is throwing.